### PR TITLE
fix: deprecated clipping + removed 'None' metrics

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR fixes: 

- a deprecation warning in both pytorch training scripts with gradient clipping function.
- a bug occurring when no boxes at all are found (edge case): the current LocalizationMetric gives back a None for both precision & IoU and this is blowing up the training script because metrics can't be logged properly.

Any feedback is welcome!

### CODE REVIEW

File: doctr/utils/metrics.py [hunk 1]
```diff
@@ -344,13 +344,13 @@ def summary(self) -> Tuple[Optional[float], Optional[float], Optional[float]]:
...
-        recall = self.matches / self.num_gts if self.num_gts > 0 else None
+        recall = self.matches / self.num_gts if self.num_gts > 0 else 0
 
         # Precision
-        precision = self.matches / self.num_preds if self.num_preds > 0 else None
+        precision = self.matches / self.num_preds if self.num_preds > 0 else 0
 
         # mean IoU
-        mean_iou = self.tot_iou / self.num_preds if self.num_preds > 0 else None
+        mean_iou = self.tot_iou / self.num_preds if self.num_preds > 0 else 0
```
Comment (fg-mindee) [inline doctr/utils/metrics.py:353]: I understand that keeping it this way would crash the script, but perhaps logging a `0` isn't a good idea either?
There is a difference between having a metric being undefined and being 0 (I'm afraid we might miss something later on)

**Review by fg-mindee — COMMENTED** [review]
Thanks for the PR!
Agree with the inplace clipping, but I added a comment for the metric
